RGPosadas/LitterRally
# TASK-79: Add friends backend

closes #277

This PR aims to implement a few backend functions to be used in the features of #273.
   - addFriend
   - removeFriend
   - getRelationships

How to test:
- Make sure you have at least 2 users in your DB.
- Run the backend
- Try to use the addFriend and removeFriend mutations and check what happens when you call the `friends` query. It should behave as expected. addFriend should be a one-sided add, equivalent to sending a request. removeFriend should be a two-way remove.
- Call the getRelationships query. This returns a list of Relationships (see added class). What I recommend is going into MySQL and modifying the rows manually and making sure that the list returned by the query changes as expected.

Examples (all from perspective of user id 1):
No relationships:
![image](https://user-images.githubusercontent.com/23544999/112736721-3164fe00-8f2b-11eb-93ea-ccff72f44763.png)

user 1 added user 2
![image](https://user-images.githubusercontent.com/23544999/112736732-535e8080-8f2b-11eb-85de-16209abde242.png)

user 2 added user 1
![image](https://user-images.githubusercontent.com/23544999/112736737-640ef680-8f2b-11eb-8f70-a88ce6f4f7a7.png)

Both users added each other
![image](https://user-images.githubusercontent.com/23544999/112736752-7852f380-8f2b-11eb-8e26-6096a8753cd6.png)

Works with multiple users too
![image](https://user-images.githubusercontent.com/23544999/112736758-899c0000-8f2b-11eb-9310-6b1546306f3a.png)
